packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/sagemaker/create-training-job.ts

```diff
@@ -324,6 +332,21 @@ export class SageMakerCreateTrainingJob extends sfn.TaskStateBase implements iam
       : {};
   }
 
+  private validateAlgorithmName(algorithmName?: string): void {
+    if (algorithmName === undefined) {
+      return;
+    }
```

Comment:
This also needs check for `Token.isUnresolved()`.

Reply:
I added check for `Token.isUnresolved()`